pkg/chartutil/capabilities.go

```diff
@@ -65,6 +76,18 @@ func (kv *KubeVersion) String() string { return kv.Version }
 // Deprecated: use KubeVersion.Version.
 func (kv *KubeVersion) GitVersion() string { return kv.Version }
 
+func ParseKubeVersion(version string) (*KubeVersion, error) {
+	sv, err := semver.NewVersion(version)
```

Comment:
This will break against versions provided by some cloud providers like GKE. They use a non-compliant semver in their kubeversion range.

Because of this, we cannot assume that the kubeversion returned by Kubernetes is semver compliant. This change will break backwards compatibility for some users.

Reply:
This is only used for parsing `--kube-version` in `helm template` so existing users should not be affected as the option doesn't even exist before this. I can inline the logic in the call site if this is a cause for confusion.

[semver.NewVersion](https://godoc.org/github.com/Masterminds/semver#NewVersion) also parses loosely so non-compliant semver should still be parsed correctly. In fact, `semverCompare` which is what's suggested to be used for comparison uses the same `semver.NewVersion` and that already works with GKE versions as when using `helm` it just provides the kubernetes version as returned by the client as-is.

This feature in [helm2](https://github.com/helm/helm/blob/a690bad98af45b015bd3da1a41f6218b1a451dbe/pkg/renderutil/render.go#L74-L80) is also implemented similarly. The main difference is just that the `Version` string is generated based on just the major/minor versions but here I build it from the parsed `String()`. I can change that to match or at least just return the original version.

Reply:
In discussing this during issue triage, we don't think this should be a problem for the use cases presented here.